docs/DeveloperGuide.md

```diff
@@ -1,25 +1,290 @@
+1. `UI` updates the user of the successful deletion of the event
+
+#### Sequence Design of Delete Function<br/>
+![](images/DeleteCommand_SequenceDiagram.png)<br/>
+
+### Clear Function
+
+#### System Architecture of the Clear Function<br/>
+
+![](images/ClearCommand_ArchitectureDiagram.png)
+
+The Architecture Diagram given above explains the high-level design of the clear command.<br/>
+* UI: The User Interface of the app<br/>
+* ClearCommand: The main logic command of the clear function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+* WhereGotTime: The main logic component of the app<br/>
+
+#### Design of the Clear Function<br/>
+
+![](images/ClearCommand_umlDiagram.png)
+
+1. `ClearCommand` gets the input from the User and gets the relevant timetable from `Timetable`
+1. `ClearCommand` then proceeds to clear all events from the timetable
+1. `ClearCommand` passes the day and cleared lesson to the `UI`
+1. `UI` updates the user of the successful clearing of events
+
+#### Sequence Design of Clear Function<br/>
+![](images/ClearCommand_SequenceDiagram.png)<br/>
+
+### List Function
+
+#### System Architecture of the List Function<br/>
+
+![](images/ListCommand_ArchitectureDiagram.png)
+
+The Architecture Diagram given above explains the high-level design of the list command.<br/>
+* UI: The User Interface of the app<br/>
+* ListCommand: The main logic command of the list function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+* WhereGotTime: The main logic component of the app<br/>
+
+#### Design of the List Function<br/>
+
+![](images/ListCommand_umlDiagram.png)
+
+1. `ListCommand` gets the input from the User and gets the relevant user's timetable(s) from `Timetable`
+1. `ListCommand` then passes the user's timetable(s) to the `UI`
+1. `UI` prints all the classes in the timetable
+
+#### Sequence Design of List Function<br/>
+![](images/ListCommand_SequenceDiagram.png)<br/>
+
+### Find Function
+
+#### System Architecture of the Find Function<br/>
+
+![](images/FindCommand_ArchitectureDiagram.png)
+
+The Architecture Diagram given above explains the high-level design of the find command.<br/>
+* UI: The User Interface of the app<br/>
+* FindCommand: The main logic command of the find function<br/>
+* Timetable: The arraylist where events that are added are stored accordingly<br/>
+* WhereGotTime: The main logic component of the app<br/>
+
+#### Design of the Find Function<br/>
+
+![](images/FindCommand_umlDiagram.png)
+
+1. `FindCommand` gets the input from the User and gets the timetable from `Timetable`
+1. `FindCommand` then searches through the timetable to find event that match the keyword entered by User
+1. `FindCommand` passes the matching events to `UI`
+1. `UI` prints the matching events
+
+#### Sequence Design of Find Function<br/>
+![](images/FindCommand_SequenceDiagram.png)<br/>
```

Comment:
<img width="596" alt="Screenshot 2020-10-29 at 4 43 37 PM" src="https://user-images.githubusercontent.com/34613421/97545395-e8556000-1a05-11eb-8b92-bf57af3301f7.png">
Does it matter if a variable was passed to FindCommand or should it be a method call with the variable inside?